### PR TITLE
Rotate image file based on Exif data

### PR DESCRIPTION
**Pull request by bparrishMines**
Fixes https://github.com/flutter/flutter/issues/19000

Copy of iOS code from https://github.com/flutter/plugins/pull/876

### CODE REVIEW

File: packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/FirebaseMlVisionPlugin.java [hunk 1]
```diff
@@ -63,11 +67,26 @@ public void onMethodCall(MethodCall call, Result result) {
 
   private FirebaseVisionImage dataToVisionImage(Map<String, Object> imageData) throws IOException {
     String imageType = (String) imageData.get("type");
+    assert imageType != null;
```
Comment (bparrishMines) [inline packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/FirebaseMlVisionPlugin.java]: Takes care of `NullPointerException` warning.

File: packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/FirebaseMlVisionPlugin.java [hunk 2]
```diff
@@ -80,12 +99,32 @@ private FirebaseVisionImage dataToVisionImage(Map<String, Object> imageData) thr
                 .setRotation(getRotation((int) metadataData.get("rotation")))
                 .build();
 
-        return FirebaseVisionImage.fromByteArray((byte[]) imageData.get("bytes"), metadata);
+        byte[] bytes = (byte[]) imageData.get("bytes");
+        assert bytes != null;
```
Comment (bparrishMines) [inline packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/FirebaseMlVisionPlugin.java]: Takes care of `NullPointerException` warning.

**Comment by sergiuiacob1** [comment]
Hey! Do you have any idea when this will be integrated into the firebase_ml_vision package?

**Comment by bparrishMines** [comment]
> Hey! Do you have any idea when this will be integrated into the `firebase_ml_vision` package?

@sergiuiacob1 published!

**Comment by p30arena** [comment]
@bparrishMines 
https://github.com/flutter/flutter/issues/28350

Should I leave the rotation to be null?

```
final FirebaseVisionImageMetadata metadata =
              FirebaseVisionImageMetadata(
                  rawFormat: image.format.raw,
                  size: Size(image.width.toDouble(), image.height.toDouble()),
                  planeData: image.planes
                      .map((currentPlane) => FirebaseVisionImagePlaneMetadata(
                          bytesPerRow: currentPlane.bytesPerRow,
                          height: currentPlane.height,
                          width: currentPlane.width))
                      .toList(),
                  rotation: ImageRotation.rotation270);
```

**Pull request merged**